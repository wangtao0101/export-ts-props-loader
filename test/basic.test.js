import compiler from './compiler.js';

describe('basic test', () => {
  test('basic loader', async () => {
    const stats = await compiler('./fixtures/reactComponent.tsx');
    expect(stats.toJson().modules[6].source).toMatchSnapshot();
  });

  test('configFile option', async () => {
    const stats = await compiler('./fixtures/reactComponent.tsx', {
      loaderOptions: { configFile: 'tsconfig_test.json' }
    });
    expect(stats.toJson().modules[6].source).toMatchSnapshot();
  });
});
