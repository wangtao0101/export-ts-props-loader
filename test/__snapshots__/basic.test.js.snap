// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic test basic loader 1`] = `
"var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
import * as React from 'react';
/**
 * My name is Test
 */
var Test = /** @class */ (function (_super) {
    __extends(Test, _super);
    function Test() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    Test.prototype.render = function () {
        return (React.createElement(\\"div\\", null, \\"aaa\\"));
    };
    Test.defaultProps = {
        text: 'text',
    };
    return Test;
}(React.PureComponent));
export default Test;
export var componentProps = [{ \\"description\\": \\"My name is Test\\", \\"displayName\\": \\"reactComponent\\", \\"props\\": { \\"visible\\": { \\"defaultValue\\": null, \\"description\\": \\"\`visible\\", \\"name\\": \\"visible\\", \\"required\\": true, \\"type\\": { \\"name\\": \\"boolean\\" } }, \\"text\\": { \\"defaultValue\\": { \\"value\\": \\"text\\" }, \\"description\\": \\"text\\", \\"name\\": \\"text\\", \\"required\\": false, \\"type\\": { \\"name\\": \\"string | undefined\\" } } } }];
"
`;

exports[`basic test configFile option 1`] = `
"var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
import * as React from 'react';
/**
 * My name is Test
 */
var Test = /** @class */ (function (_super) {
    __extends(Test, _super);
    function Test() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    Test.prototype.render = function () {
        return (React.createElement(\\"div\\", null, \\"aaa\\"));
    };
    Test.defaultProps = {
        text: 'text',
    };
    return Test;
}(React.PureComponent));
export default Test;
export var componentProps = [{ \\"description\\": \\"My name is Test\\", \\"displayName\\": \\"reactComponent\\", \\"props\\": { \\"visible\\": { \\"defaultValue\\": null, \\"description\\": \\"\`visible\\", \\"name\\": \\"visible\\", \\"required\\": true, \\"type\\": { \\"name\\": \\"boolean\\" } }, \\"text\\": { \\"defaultValue\\": { \\"value\\": \\"text\\" }, \\"description\\": \\"text\\", \\"name\\": \\"text\\", \\"required\\": false, \\"type\\": { \\"name\\": \\"string | undefined\\" } } } }];
"
`;

exports[`basic test module option 1`] = `
"var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
import * as React from 'react';
/**
 * My name is Test
 */
var Test = /** @class */ (function (_super) {
    __extends(Test, _super);
    function Test() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    Test.prototype.render = function () {
        return (React.createElement(\\"div\\", null, \\"aaa\\"));
    };
    Test.defaultProps = {
        text: 'text',
    };
    return Test;
}(React.PureComponent));
export default Test;
module.exports.componentProps = [{ \\"description\\": \\"My name is Test\\", \\"displayName\\": \\"reactComponent\\", \\"props\\": { \\"visible\\": { \\"defaultValue\\": null, \\"description\\": \\"\`visible\\", \\"name\\": \\"visible\\", \\"required\\": true, \\"type\\": { \\"name\\": \\"boolean\\" } }, \\"text\\": { \\"defaultValue\\": { \\"value\\": \\"text\\" }, \\"description\\": \\"text\\", \\"name\\": \\"text\\", \\"required\\": false, \\"type\\": { \\"name\\": \\"string | undefined\\" } } } }];
"
`;
